hathor/verification/block_verifier.py

```diff
@@ -52,25 +60,45 @@ def verify(self, block: Block) -> None:
     def verify_without_storage(self, block: Block) -> None:
         """ Run all verifications that do not need a storage.
         """
-        block.verify_without_storage()
+        self.verify_pow(block)
+        self.verify_no_inputs(block)
+        self.verify_outputs(block)
+        self.verify_data(block)
+        self.verify_sigops_output(block)
 
     def verify_height(self, block: Block) -> None:
         """Validate that the block height is enough to confirm all transactions being confirmed."""
-        block.verify_height()
+        meta = block.get_metadata()
+        assert meta.height is not None
+        assert meta.min_height is not None
+        if meta.height < meta.min_height:
+            raise RewardLocked(f'Block needs {meta.min_height} height but has {meta.height}')
 
     def verify_weight(self, block: Block) -> None:
         """Validate minimum block difficulty."""
-        block.verify_weight()
+        block_weight = daa.calculate_block_difficulty(block)
```

Comment:
Rename to `min_block_weight`? I think having both `block.weight` and `block_weight` is error prone.

Reply:
Maybe add it as a separate commit in this PR? Or else I think we should get this new PR merged before moving forward with the other parts.

Reply:
Sure, done in https://github.com/HathorNetwork/hathor-core/pull/798/commits/61cc04be1b99a23a5d2c899b89fe512fa1a7228c

Reply:
Thanks!